src/components/Api.js

```diff
@@ -0,0 +1,89 @@
+export default class Api {
+  constructor({ baseUrl, headers }) {
+    this._baseUrl = baseUrl;
+    this._headers = headers;
+  }
+
+  _request(url, options) {
+    return fetch(url, options)
+      .then(this._checkResponse)
```

Comment:
`      .then(this._checkResponse)
`

You don't have this method. Please, use `renderResult` instead of it


```js

  _request(url, options) {
    return fetch(url, options)
      .then(this.renderResult)
  }
```